src/components/SearchField.tsx

```diff
@@ -0,0 +1,54 @@
+import { Cross2Icon, MagnifyingGlassIcon } from '@radix-ui/react-icons'
+import { IconButton, TextField } from '@radix-ui/themes'
+import { RootProps } from '@radix-ui/themes/dist/cjs/components/text-field'
+import { forwardRef, useCallback, useImperativeHandle, useRef } from 'react'
+
+export interface SearchFieldHandle {
+  focus(): void
+  clear(): void
+}
+
+type SearchFieldProps = Omit<RootProps, 'onChange'> & {
+  filter: string
+  onChange: (filter: string) => void
+}
+
+export const SearchField = forwardRef<SearchFieldHandle, SearchFieldProps>(
+  function SearchField({ filter, onChange, ...inputProps }, ref) {
+    const inputRef = useRef<HTMLInputElement | null>(null)
+
+    const clear = useCallback(() => {
+      onChange('')
+    }, [onChange])
+
+    useImperativeHandle(ref, () => {
```

Comment:
The first time I've ever found `useImperativeHandle` to be helpful. It gives the user the same control over the input as if they had a `ref` to the raw element itself.

Reply:
TIL 💡